pkgs/tools/graphics/ldgallery/viewer/generate.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nodePackages.node2nix
+
+# TODO: merge with other node packages once support for npm projects in sub-directories
+# is added to npm: https://github.com/npm/cli/issues/528
+
+# Development dependencies are required for this Vue application to build
+
+TAG="v1.0"
+
+wget https://github.com/pacien/ldgallery/raw/$TAG/viewer/package.json
+wget https://github.com/pacien/ldgallery/raw/$TAG/viewer/package-lock.json
+
+node2nix \
+  --nodejs-12 \
+  --node-env ../../../../development/node-packages/node-env.nix \
+  --development \
+  --input ./package.json \
+  --lock ./package-lock.json \
+  --output node-packages.nix \
+  --composition node-composition.nix
+
+rm package.json package-lock.json
```

Comment:
Here I have two problems that prevent me from simply adding the "viewer" part to the common list of node packages in pkgs/development/node-packages/node-packages-12.json:

* npm doesn't allow importing projects from a sub-directory in a repository (https://github.com/npm/cli/issues/528), so I have to manually extract the package.json and package-lock.json files and run node2nix on those.
* Building the package (which uses TypeScript and WebPack) also requires downloading development dependencies with the "--development" option when calling node2nix, which isn't used for the common list of node packages.

Keeping this package separate with a generate.sh script seems to be an acceptable option to work around those issues.